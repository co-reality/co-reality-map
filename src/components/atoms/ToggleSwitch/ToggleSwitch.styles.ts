import styled, { css } from "styled-components";

export const Label = styled.label`
  display: inline-block;
  margin: 0 1rem;

  position: relative;
`;

type WrapperProps = {
  large?: boolean;
};
const defaultToggle = css`
  ${Label} {
    width: 40px;
    height: 23px;
  }
`;
const largeToggle = css`
  ${Label} {
    width: 6rem;
    height: 2.5rem;
  }
`;
export const Wrapper = styled.div<WrapperProps>`
  display: flex;
  align-items: center;

  ${({ large }) => (large ? largeToggle : defaultToggle)};
`;

type SliderType = {
  checked: boolean;
  large?: boolean;
};
const defaultSlider = css`
  background-color: #f6e652;
`;
const checkedSlider = css<SliderType>`
  background-color: #4bcc4b;

  &::before {
    transform: ${({ large }) =>
      large ? "translateX(3.5rem)" : "translateX(17px)"};
  }
`;
export const Slider = styled.span`
  ${(props: SliderType) => (props.checked ? checkedSlider : defaultSlider)};

  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;

  border-radius: 22px;

  cursor: pointer;

  transition: 0.4s;

  &::before {
    content: "";

    width: ${({ large }) => (large ? "2.1rem" : "17px")};
    height: ${({ large }) => (large ? "2.1rem" : "17px")};

    position: absolute;
    bottom: 3px;
    left: 3px;

    background-color: white;
    border-radius: 50%;

    transition: transform ease 0.4s;
  }
`;

export const Input = styled.input.attrs({
  hidden: true,
  type: "checkbox",
})``;

export const Text = styled.span`
  font-size: 1.3rem;
`;
