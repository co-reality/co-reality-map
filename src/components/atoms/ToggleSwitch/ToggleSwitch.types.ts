export interface SwitchProps {
  name: string;
  forwardRef?: any;
  isChecked?: boolean;
  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
  withText?: boolean;
  isLarge?: boolean;
}
