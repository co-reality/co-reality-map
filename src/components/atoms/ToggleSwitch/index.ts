/**
 * @deprecated Use components/atoms/Toggler instead
 */
export { default } from "./ToggleSwitch";
