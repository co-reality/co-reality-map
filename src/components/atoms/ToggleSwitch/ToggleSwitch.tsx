import React, { useState } from "react";

// Typings
import { SwitchProps } from "./ToggleSwitch.types";

// Styles
import * as S from "./ToggleSwitch.styles";

const ToggleSwitch: React.FC<SwitchProps> = ({
  name,
  forwardRef,
  isChecked = false,
  onChange,
  withText = false,
  isLarge = false,
}) => {
  const [checked, setChecked] = useState<boolean>(isChecked);

  const handleOnChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    if (onChange) {
      onChange(e);
    }

    return setChecked(e.target.checked);
  };

  return (
    <S.Wrapper large={isLarge}>
      {withText && <S.Text>OFF</S.Text>}
      <S.Label>
        <S.Input
          name={name}
          ref={forwardRef}
          onChange={(e) => handleOnChange(e)}
          checked={checked}
        />

        <S.Slider checked={checked} large={isLarge} />
      </S.Label>
      {withText && <S.Text>ON</S.Text>}
    </S.Wrapper>
  );
};

export default ToggleSwitch;
