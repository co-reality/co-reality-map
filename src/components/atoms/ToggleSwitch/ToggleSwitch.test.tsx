import React from "react";
import renderer from "react-test-renderer";

// Test component
import ToggleSwitch from "./ToggleSwitch";

describe("<ToggleSwitch />", () => {
  test("renders correctly", () => {
    const tree = renderer.create(<ToggleSwitch name="mySwitch" />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
