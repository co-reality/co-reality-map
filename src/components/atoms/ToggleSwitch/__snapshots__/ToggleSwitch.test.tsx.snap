// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleSwitch /> renders correctly 1`] = `
<label
  className="sc-bdfBwQ jGgSFq"
>
  <input
    checked={false}
    className="sc-dlfnbm"
    hidden={true}
    name="mySwitch"
    onChange={[Function]}
    type="checkbox"
  />
  <span
    checked={false}
    className="sc-gsTCUz HBMiV"
  />
</label>
`;
